crates/noirc_evaluator/src/ssa/optim.rs

```diff
@@ -17,7 +17,10 @@ use std::{
 
 // Performs constant folding, arithmetic simplifications and move to standard form
 // Modifies ins.mark with whether the instruction should be deleted, replaced, or neither
-pub fn simplify(ctx: &mut SsaContext, ins: &mut node::Instruction) {
+pub fn simplify(ctx: &mut SsaContext, ins: &mut Instruction) {
+    if ins.is_deleted() {
+        return;
+    }
     //1. constant folding
     let new_id = match ins.evaluate(ctx) {
         NodeEval::Const(c, t) => ctx.get_or_create_const(c, t),
```

Comment:
While we're changing things, I noticed this is the same as the to_index function. Can we use that here?

Reply:
Done